crates/astria-account-monitor/src/config.rs

```diff
@@ -0,0 +1,184 @@
+use std::{
+    borrow::Cow,
+    collections::HashSet,
+    fmt::Display,
+    hash::Hash,
+    str::FromStr,
+};
+
+use astria_core::primitive::v1::asset::Denom;
+use astria_eyre::eyre::{
+    self,
+};
+use sequencer_client::Address;
+use serde::Deserialize;
+
+#[derive(Debug, Deserialize)]
+/// The high-level config for creating an astria-account-monitor service.
+pub struct Config {
+    /// Log level. One of debug, info, warn, or error
+    pub log: String,
+
+    /// Address of the ABCI server for the sequencer chain
+    pub sequencer_abci_endpoint: String,
+
+    /// The chain ID of the sequencer chain
+    pub sequencer_chain_id: String,
+
+    /// The addresses of the sequencer chain to monitor.
+    pub sequencer_accounts: SequencerAccountsToMonitor,
+
+    /// The asset ID of the sequencer chain to monitor.
+    pub sequencer_asset: Asset,
+
+    /// Sequencer block time in milliseconds
+    pub query_interval_ms: u64,
+
+    /// Forces writing trace data to stdout no matter if connected to a tty or not.
+    pub force_stdout: bool,
+
+    /// Disables writing trace data to an opentelemetry endpoint.
+    pub no_otel: bool,
+
+    /// The endpoint which will be listened on for serving prometheus metrics
+    pub metrics_http_listener_addr: String,
+}
+
+#[derive(Debug, Clone)]
+pub struct Asset {
+    /// The asset ID of the sequencer chain to monitor.
+    pub asset: Denom,
+}
+
+impl<'de> Deserialize<'de> for Asset {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: serde::Deserializer<'de>,
+    {
+        let value = String::deserialize(deserializer)?;
+        let asset = value.parse().map_err(serde::de::Error::custom)?;
+
+        Ok(Self {
+            asset,
+        })
+    }
+}
+
+impl Hash for Asset {
+    fn hash<H: std::hash::Hasher>(&self, state: &mut H) {
+        self.asset.hash(state);
+    }
+}
+
+impl Display for Asset {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "{}", self.asset)
+    }
+}
+
+#[derive(Debug, Clone)]
+pub struct SequencerAccountsToMonitor(Vec<Account>);
+
+impl<'de> Deserialize<'de> for SequencerAccountsToMonitor {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: serde::Deserializer<'de>,
+    {
+        let value = String::deserialize(deserializer)?;
+
+        let accounts: Result<Vec<Account>, _> = value
+            .split(',')
+            .map(str::trim)
+            .filter(|s| !s.is_empty())
```

Comment:
I wouldn't filter these. If somebody provides `astriaxxxx,,astriaxxxxx` then that empty field is just a wrong address.